scripts/parse.py

```diff
@@ -170,36 +189,48 @@ def parsesalt(infile, outfile, nestinglevel):
     salt = bytearray2hex(readbytearray(infile, 16))
     outfile.write(salt + ' (salt)\n')
 
-def parsesizedcontent(infile, outfile, nestinglevel, desc):
+def parsesizedcontent(infile, outfile, nestinglevel, enumid):
     length = readshort(infile)
     text = readstring(infile, length)
-    outfile.write('"%s"%s\n' % (text, desc2suffix(desc)))
+    valuedescSet = None
+    # See if text parses to int, it might be a known constant value
+    try:
+        value = int(text, 0)
+        valuedescSet = enumwithvalue2valuedesc(knownTAConstants, enumid, value)
+    except ValueError:
+        pass
+    outfile.write('"%s"%s\n' % (text, desc2suffix(enum2desc(enumid), valuedescSet)))
 
 def parsearrayofenumblockarrays(infile, outfile, nestinglevel):
     size = readshort(infile)
     outfile.write('arrayofenumblockarrays size %d\n' % size)
     for i in range(size):
         parseenumblockarray(infile, outfile, nestinglevel + 1, True, prefix=index2prefix(i))
 
-def parseonebyte(infile, outfile, nestinglevel, desc):
+def parseonebyte(infile, outfile, nestinglevel, enumid):
     value = readbyte(infile)
-    outfile.write('%02X%s\n' % (value, desc2suffix(desc)))
+    valuedescSet = enumwithvalue2valuedesc(knownTAConstants, enumid, value)
+    outfile.write('%02X%s\n' % (value, desc2suffix(enum2desc(enumid), valuedescSet)))
     
-def parsetwobytes(infile, outfile, nestinglevel, desc):
+def parsetwobytes(infile, outfile, nestinglevel, enumid):
     value = readshort(infile)
-    outfile.write('%04X%s\n' % (value, desc2suffix(desc)))
+    valuedescSet = enumwithvalue2valuedesc(knownTAConstants, enumid, value)
+    outfile.write('%04X%s\n' % (value, desc2suffix(enum2desc(enumid), valuedescSet)))
 
-def parsethreebytes(infile, outfile, nestinglevel, desc):
+def parsethreebytes(infile, outfile, nestinglevel, enumid):
     value = bytearray2hex(readbytearray(infile, 3))
-    outfile.write('%s%s\n' % (value, desc2suffix(desc)))
+    valuedescSet = enumwithvalue2valuedesc(knownTAConstants, enumid, value)
+    outfile.write('%s%s\n' % (value, desc2suffix(enum2desc(enumid), valuedescSet)))
 
-def parsefourbytes(infile, outfile, nestinglevel, desc):
+def parsefourbytes(infile, outfile, nestinglevel, enumid):
     value = readlong(infile)
-    outfile.write('%08X%s\n' % (value, desc2suffix(desc)))
+    valuedescSet = enumwithvalue2valuedesc(knownTAConstants, enumid, value)
+    outfile.write('%08X%s\n' % (value, desc2suffix(enum2desc(enumid), valuedescSet)))
 
-def parseeightbytes(infile, outfile, nestinglevel, desc):
+def parseeightbytes(infile, outfile, nestinglevel, enumid):
     value = bytearray2hex(readbytearray(infile, 8))
-    outfile.write('%s%s\n' % (value, desc2suffix(desc)))
+    valuedescSet = enumwithvalue2valuedesc(knownTAConstants, enumid, value)
```

Comment:
bytearray2hex returns a string of the form "12 34 56 78 9A BC", so looking that up in TA constants is never going to find anything

Reply:
Ah right. We shouldn't be seeing constants in an eight byte enum output anyway I'd expect...